### PR TITLE
fix: drop column when focusing on data cell

### LINKED ISSUE #50: Column Ordering - grab and release bug
grabbing a column and then keyboard navigate to data cell should drop column

### PR DESCRIPTION
**Pull request by steve-haar**
fixes #50

### CODE REVIEW

File: grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering.service.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable()
+export class CustomClrColumnOrderingService {
+  grabbedColumn = new BehaviorSubject<any>(null);
```
Comment (kevinbuhmann) [inline grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering.service.ts:6]: Can this be `readonly`?

Comment (steve-haar) [reply grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering.service.ts]: yes, it should be

File: grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering-column.directive.ts [hunk 2]
```diff
@@ -1,44 +1,65 @@
...
     cdkDrag: CdkDrag
   ) {
     cdkDrag.previewContainer = 'parent';
   }
 
+  private subscription: Subscription | null = null;
+
   @Input() customClrColumnOrderingColumn: any;
   @Input() columnIndex!: number;
+  _isGrabbed = false;
```
Comment (kevinbuhmann) [inline grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering-column.directive.ts:37]: I think this could be removed.

Comment (steve-haar) [reply grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering-column.directive.ts]: fixed

**Review by kevinbuhmann — APPROVED** [review]

File: grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering-column.directive.ts [hunk 3]
```diff
@@ -1,44 +1,59 @@
...
     private readonly elementRef: ElementRef,
+    private readonly columnOrderingService: CustomClrColumnOrderingService,
+    private readonly changeDetectorRef: ChangeDetectorRef,
+    private readonly column: ClrDatagridColumn,
     cdkDrag: CdkDrag
   ) {
     cdkDrag.previewContainer = 'parent';
   }
 
+  private subscription: Subscription | null = null;
```
Comment (kevinbuhmann) [inline grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering-column.directive.ts:33]: I think this should be `private subscription: Subscription | undefined;` for consistency.

**Review by kevinbuhmann — CHANGES_REQUESTED** [review]

File: grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering-column.directive.ts (hunk 3 above)
Comment (steve-haar) [reply grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering-column.directive.ts]: fixed

**Review by kevinbuhmann — APPROVED** [review]

**Pull request merged**